packages/vertexai/src/requests/stream-reader.ts

```diff
@@ -184,7 +185,13 @@ export function aggregateResponses(
           }
           const newPart: Partial<Part> = {};
           for (const part of candidate.content.parts) {
-            if (part.text) {
+            if (part.text !== undefined) {
+              // The backend can send empty text parts. If these are sent back
+              // (e.g. in chat history), the backend will respond with an error.
+              // To prevent this, ignore empty text parts.
+              if (part.text === '') {
+                continue;
+              }
```

Comment:
I can't select it but I'm a little worried about line 201 where I think we _create_ a part with an empty text string if the `newPart` hasn't had any properties added so far. I don't really remember why we did that. But now that's a possible source of a part with an empty text string, which I think we are trying to forbid.

Reply:
Good point. I removed that.

Our part types all have at least one required property, so I assume that means the backend should never send a Part with no properties. Do you think we should throw an error in this case?

Reply:
It seems like we should. I don't know what the fallback would be. I guess the fallback here was a Part with an empty text string, but that's an error, so I can't think of what else. Anything else (like " ") seems like inventing content that isn't there.